Detectors/PHOS/simulation/include/PHOSSimulation/Detector.h

```diff
@@ -0,0 +1,179 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_PHOS_DETECTOR_H_
+#define ALICEO2_PHOS_DETECTOR_H_
+
+#include "DetectorsBase/Detector.h"
+#include "PHOSBase/Hit.h"
+#include "MathUtils/Cartesian3D.h"
+#include "RStringView.h"
+#include "Rtypes.h"
+
+#include <vector>
+#include <map>
+
+class FairVolume;
+
+namespace o2
+{
+namespace PHOS
+{
+class Hit;
+class Geometry;
+class GeometryParams;
+
+///
+/// \class Detector
+/// \brief Detector class for the PHOS detector
+///
+/// The detector class handles the implementation of the PHOS detector
+/// within the virtual Monte-Carlo framework and the simulation of the
+/// PHOS detector up to hit generation
+class Detector : public o2::Base::DetImpl<Detector>
+{
+ public:
+  //PHOS materials/media
+  enum { ID_PWO = 0, ID_CPVSC = 1, ID_AL = 2, ID_TYVEK = 3, ID_POLYFOAM = 4, ID_TITAN = 5, ID_APD = 6,ID_THERMOINS = 7,
+         ID_TEXTOLIT = 8,     ID_CUPPER = 10, ID_PRINTCIRC = 11, ID_CO2 = 12, ID_FE = 13, ID_FIBERGLASS = 14,
+         ID_CABLES = 15, ID_AIR=16 };
+
+  ///
+  /// Default constructor
+  ///
+  Detector() = default;
+
+  ///
+  /// Main constructor
+  ///
+  /// \param[in] isActive Switch whether detector is active in simulation
+  Detector(Bool_t isActive);
+
+  ///
+  /// Destructor
+  ///
+  ~Detector() override = default;
+
+  ///
+  /// Initializing detector
+  ///
+  void Initialize() final;
+
+  ///
+  /// Processing hit creation in the PHOS crystalls
+  ///
+  /// \param[in] v Current sensitive volume
+  Bool_t ProcessHits(FairVolume* v = nullptr) final ;
+
+  ///
+  /// Add PHOS hit
+  /// Internally adding hits coming from the same track
+  ///
+  /// \param[in] trackID Index of the track in the MC stack first entered PHOS
+  /// \param[in] detID Index of the detector (cell) for which the hit is created
+  /// \param[in] pos Position vector of the particle first entered PHOS
+  /// \param[in] mom Momentum vector of the particle first entered PHOS
+  /// \param[in] totE Total energy of the particle entered PHOS
+  /// \param[in] time Time of the hit
+  /// \param[in] energyloss Energy deposited in this step
+  ///
+  Hit* AddHit(Int_t trackID, Int_t detID,
+                      const Point3D<float>& pos, const Vector3D<float>& mom, Double_t totE, Double_t time, Double_t eLoss) ;
```

Comment:
Could you try to run `clang-format` on all the new files? This should provide the correct formatting.